src/FastRandom.cpp

```diff
@@ -8,8 +8,11 @@
 
 #include <random>
 
-#if defined (__GNUC__)
-	#define ATTRIBUTE_TLS static __thread
+#if defined (ANDROID)
+	// Tycho: I'm sorry, I'm sorry, I'm so, so, sorry, but please blame the NDK developers for lack of, thread_local
+	#define ATTRIBUTE_TLS static
```

Comment:
I am intending for this to be an experimental thing. Of course there will be bugs, but there are loads of them anyway, and the compiler doesn't seem to be doing anything untoward.

In an ideal situation, everything would be standards compliant, but since I've no time to reimplement <atomic> nor wish to pollute the main repository with more Android artefacts, it's gonna be a case of buggy mobile builds versus none at all.

Does that sit comfortable on your conscience??

...unless you want to do something code-y about it?


On 21 Dec 2015, at 17:28, "worktycho" <notifications@github.com> wrote:

> In src/FastRandom.cpp:
> 
> > @@ -8,8 +8,11 @@
> >  
> >  #include <random>
> >  
> > -#if defined (__GNUC__)
> > -	#define ATTRIBUTE_TLS static __thread
> > +#if defined (ANDROID)
> > +	// Tycho: I'm sorry, I'm sorry, I'm so, so, sorry, but please blame the NDK developers for lack of, thread_local
> > +	#define ATTRIBUTE_TLS static
> Why not just replace cCriticalSection with a noop? It will offer a major performance boost on these low power devices? Because we need thread safety. Any instance of a race condition is grounds for the compiler to do whatever it feels like.
> 
> —
> Reply to this email directly or view it on GitHub.
>